lib/github-pages-health-check/domain.rb

```diff
@@ -449,8 +449,11 @@ def enforces_https?
       def https_eligible?
         # Can't have any IP's which aren't GitHub's present.
         return false if non_github_pages_ip_present?
-        # Must be a CNAME or point to our IPs.
 
+        # Can't have underscores in the domain name
+        return false if host.include?("_")
```

Comment:
Should this invalidate the domain name as well, or just HTTPS eligibility? Does GitHub have many customer custom domains with underscores?

Reply:
Hello there!

We do have people on HTTP only and with a `_` in their domain name unfortunately 🙈. Some are using Cloudflare which does issue certs, but Let's Encrypt does not.

Reply:
Gotta love the edge cases! 😄

Reply:
We are full of them!